src/Test/Utilities/Shared/Assert/EqualityUtil`1.cs

```diff
@@ -161,7 +169,7 @@ private void ObjectEquals3()
             var allValues = _equalityUnits.SelectMany(x => x.AllValues);
             foreach (var value in allValues)
             {
-                Assert.NotEqual((object)42, value);
+                Assert.False(value.Equals((object)42));
```

Comment:
@jasonmalinowski the intent of this check was to verify the behavior of `object.Equals(object)`.  It seemed wrong to rely on this being tested as an implementation detail of `Assert.NotEqual` (and in fact it doesn't guarantee this).  Much more direct to just call it directly.